addon/utils/collection-action.ts

```diff
@@ -14,24 +14,38 @@ export interface CollectionOperationOptions<IN, OUT> {
 }
 
 export default function collectionOp<IN = any, OUT = any>(options: CollectionOperationOptions<IN, OUT>) {
-  return function runCollectionOp(this: Model, payload: IN): Promise<OUT> {
-    const model: Model = this;
-    const recordClass = _getModelClass(model);
+  return async function runCollectionOp(this: Model, payload: IN): Promise<OUT> {
```

Comment:
@Turbo87 called this out to me and I agree with the reasoning -- it might be dangerous to use `async/await` and force a user of this library to transpile that syntax and possibly ship with `regenerator`. It would probably be safer to keep this using Promises "directly" instead.

Reply:
I'll bring the old behaviour back - Good to know for the future, thanks!